consensus/polybft/governance_manager.go

```diff
@@ -0,0 +1,411 @@
+			"epoch", req.Epoch, "block", req.FullBlock.Block.Number(), "lastGottenBlock", lastBlock)
+
+		// even if there were no governance events in block, mark it as last processed
+		return g.state.GovernanceStore.insertLastProcessed(req.FullBlock.Block.Number())
+	}
+
+	g.logger.Debug("Post block - Getting governance events done.", "epoch", req.Epoch,
+		"block", req.FullBlock.Block.Number(), "eventsNum", numOfEvents)
+
+	// this will also update the last processed block
+	return g.state.GovernanceStore.insertGovernanceEvents(
+		req.Epoch,
+		req.FullBlock.Block.Number(),
+		governanceEvents)
+}
+
+// parseGovernanceEvent parses provided log to correct governance event
+func parseGovernanceEvent(h *types.Header, log *ethgo.Log) (contractsapi.EventAbi, bool, error) {
+	var (
+		checkpointIntervalEvent  contractsapi.NewCheckpointBlockIntervalEvent
+		epochSizeEvent           contractsapi.NewEpochSizeEvent
+		epochRewardEvent         contractsapi.NewEpochRewardEvent
+		minValidatorSetSizeEvent contractsapi.NewMinValidatorSetSizeEvent
+		maxValidatorSetSizeEvent contractsapi.NewMaxValidatorSetSizeEvent
+		withdrawalPeriodEvent    contractsapi.NewWithdrawalWaitPeriodEvent
+		blockTimeEvent           contractsapi.NewBlockTimeEvent
+		blockTimeDriftEvent      contractsapi.NewBlockTimeDriftEvent
+		votingDelayEvent         contractsapi.NewVotingDelayEvent
+		votingPeriodEvent        contractsapi.NewVotingPeriodEvent
+		proposalThresholdEvent   contractsapi.NewProposalThresholdEvent
+		sprintSizeEvent          contractsapi.NewSprintSizeEvent
+	)
+
+	parseEvent := func(event contractsapi.EventAbi) (contractsapi.EventAbi, bool, error) {
+		doesMatch, err := event.ParseLog(log)
+
+		return event, doesMatch, err
+	}
+
+	switch log.Topics[0] {
+	case checkpointIntervalEvent.Sig():
+		return parseEvent(&checkpointIntervalEvent)
+	case epochSizeEvent.Sig():
+		return parseEvent(&epochSizeEvent)
+	case epochRewardEvent.Sig():
+		return parseEvent(&epochRewardEvent)
+	case minValidatorSetSizeEvent.Sig():
+		return parseEvent(&minValidatorSetSizeEvent)
+	case maxValidatorSetSizeEvent.Sig():
+		return parseEvent(&maxValidatorSetSizeEvent)
+	case withdrawalPeriodEvent.Sig():
+		return parseEvent(&withdrawalPeriodEvent)
+	case blockTimeEvent.Sig():
+		return parseEvent(&blockTimeEvent)
+	case blockTimeDriftEvent.Sig():
+		return parseEvent(&blockTimeDriftEvent)
+	case votingDelayEvent.Sig():
+		return parseEvent(&votingDelayEvent)
+	case votingPeriodEvent.Sig():
+		return parseEvent(&votingPeriodEvent)
+	case proposalThresholdEvent.Sig():
+		return parseEvent(&proposalThresholdEvent)
+	case sprintSizeEvent.Sig():
+		return parseEvent(&sprintSizeEvent)
+	default:
+		return nil, false, errUnknownGovernanceEvent
+	}
+}
+
+// unmarshalGovernanceEvent unmarshals given raw event to desired type
+func unmarshalGovernanceEvent[T contractsapi.EventAbi](rawEvent []byte) (T, error) {
+	var event T
+
+	err := json.Unmarshal(rawEvent[32:], &event)
```

Comment:
Extract 32 to const?

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/1749/commits/6bcffbbd344e5db00b863d22c1872ace2a23a9f6